python-lib/sharepoint_client.py

```diff
@@ -262,9 +262,21 @@ def create_folder(self, full_path):
         response = self.session.post(
             self.get_add_folder_url(full_path)
         )
-        self.assert_response_ok(response, calling_method="create_folder")
         return response
 
+    def create_path(self, file_full_path):
+        full_path, filename = os.path.split(file_full_path)
+        tokens = full_path.split("/")
+        path = ""
+        previous_status = None
+        for token in tokens:
+            previous_path = path
+            path = get_lnt_path(path + "/" + token)
+            response = self.create_folder(path)
+            status_code = response.status_code
+            if previous_status == 403 and status_code == 404:
+                logger.error("Could not create folder for '{}'. Check your write permission for the folder {}.".format(path, previous_path))
```

Comment:
`create_path` is called every time a file is uploaded (because checking the whole path existence takes as many calls as to building it, it is on average shorter to just build it). This mean the path might already exists, regardless of errors during create_path. That's why this method does not raise exception: errors at this stage might not be an issue, and if it is, we'll get a 404 while doing the final file upload and raise an error then.

Reply:
Sounds fine - my only issue was that the 404 at the end might be confusing if that wasn't the real cause of the problem. Anyway like I say it's very much an edge case, ok as is.